packages/admin/cms-admin/src/generator/future/generator.ts

```diff
@@ -42,7 +42,7 @@ export type GridColumnConfig<T> = (
     | { type: "dateTime" }
     | { type: "staticSelect"; values?: string[] }
     | { type: "block"; block: ImportReference }
-) & { name: keyof T; headerName?: string; width?: number };
+) & { name: keyof T; headerName?: string; width?: number; minWidth?: number; maxWidth?: number; flex?: number };
```

Comment:
Should we add JSDoc comments linking to the respective values in the MUI docs here?

Reply:
Ideally I would have picked the values from `GridColDef` (e.g. `import { GridColDef } from "@mui/x-data-grid"`) but wasn't sure if that's a good idea, because there are different data-grid packages, depending on the project, not just `@mui/x-data-grid`.

Do you have an example of how this would work with JSDoc?

Reply:
Something like this:

```ts
export type GridColumnConfig<T> = (
  ...
) & {
    name: keyof T;
    headerName?: string;
    /**
     * Column width
     * @see https://mui.com/x/api/data-grid/grid-col-def/#properties
     */
    width?: number;
};
```

But I noticed that linking to specific props is not possible 😢

Reply:
Ok, it looks like both `data-grid-pro` and `data-grid-premium` have `@mui/x-data-grid` as a dependency and use that packages `GridColDef` type, so it should be fine if we pick it from there. 

Then it should be clear in the IDE where the type comes from.